apollo-router/tests/integration/query_planner.rs

```diff
@@ -0,0 +1,466 @@
+use std::path::PathBuf;
+
+use crate::integration::common::graph_os_enabled;
+use crate::integration::IntegrationTest;
+
+const PROMETHEUS_METRICS_CONFIG: &str = include_str!("telemetry/fixtures/prometheus.router.yaml");
+const LEGACY_QP: &str = "experimental_query_planner_mode: legacy";
+const NEW_QP: &str = "experimental_query_planner_mode: new";
+const BOTH_QP: &str = "experimental_query_planner_mode: both";
+const BOTH_BEST_EFFORT_QP: &str = "experimental_query_planner_mode: both_best_effort";
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_legacy_qp() {
+    let mut router = IntegrationTest::builder()
+        .config(LEGACY_QP)
+        .supergraph("../examples/graphql/local.graphql")
+        .build()
+        .await;
+    router.start().await;
+    router.assert_started().await;
+    router.execute_default_query().await;
+    router.graceful_shutdown().await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_new_qp() {
+    let mut router = IntegrationTest::builder()
+        .config(NEW_QP)
+        .supergraph("../examples/graphql/local.graphql")
+        .build()
+        .await;
+    router.start().await;
+    router
+        .assert_log_contains(
+            "could not create router: \
+             The supergraph schema failed to produce a valid API schema: \
+             Supergraphs composed with federation version 1 are not supported.",
+        )
+        .await;
+    router.assert_shutdown().await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_both_qp() {
+    let mut router = IntegrationTest::builder()
+        .config(BOTH_QP)
+        .supergraph("../examples/graphql/local.graphql")
+        .build()
+        .await;
+    router.start().await;
+    router
+        .assert_log_contains(
+            "could not create router: \
+             The supergraph schema failed to produce a valid API schema: \
```

Comment:
The mention of API schema here is not quite correct. It comes from `apollo-router/src/error.rs`:

```
impl From<FederationError> for ServiceBuildError {
    fn from(err: FederationError) -> Self {
        ServiceBuildError::ApiSchemaError(err)
    }
}
```

QP initialization is the only place where this enum variant is used (since API schema generation has moved to `apollo_router::spec::Schema` creation), so it should be renamed to `QueryPlannerInit` or something, and the message reworded